### PR TITLE
CVE-2016-8740

### PR DESCRIPTION
**Pull request by 17mgeffert**
httpd issue with denial of service

### CODE REVIEW

File: cves/CVE-2016-8740.yml [hunk 1]
```diff
@@ -5,6 +5,8 @@ CWE_instructions: |
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
 CWE:
+  The mod_http2 module in the Apache HTTP Server 2.4.17 through 2.4.23, when the Protocols configuration includes h2 or h2c, does not restrict request-header length, which allows remote attackers to cause a denial of service
```
Comment (mesternefeld) [inline cves/CVE-2016-8740.yml]: I'm a little confused with what this is. It should be the CWE of the CVE. Commonly in the format CWE-400 or just simply 400

File: cves/CVE-2016-8740.yml [hunk 2]
```diff
@@ -49,6 +51,10 @@ description_instructions: |
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
 description:
+  A specific module, the modhttp2 module did not offer data length
+  restriction for the request headers. This means that malicious sources could potentially
+  send through large amounts of information, and block out threads, stopping others
+  from being able to use it.
```
Comment (mev1482) [inline cves/CVE-2016-8740.yml]: Is a denial of service attack could be accomplished? May want to state that explicitly if that is the case.

File: cves/CVE-2016-8740.yml [hunk 3]
```diff
@@ -68,11 +74,17 @@ fixes_vcc_instructions: |
...
+        Reported by Naveen Tiwari <naveen.tiwari@asu.edu> and CDF/SEFCOM at Arizona State University
+        git-svn-id https://svn.apache.org/repos/asf/httpd/httpd/trunk@1772576 13f79535-47bb-0310-9956-ffa450edef68"
    - commit:
      note:
 vccs:
-  - commit:
+  - commit: b87719bebc698f0c7da44cc57b0ecd101c5f0578
     note:
+      mod_http2 fixes timeout vs. keepalive timeout use when closing strems on flush, adding pseudo header length check against limit_req_line
+      git-svn-id https://svn.apache.org/repos/asf/httpd/httpd/trunk@1744415 13f79535-47bb-0310-9956-ffa450edef68"
```
Comment (mev1482) [inline cves/CVE-2016-8740.yml:87]: May want to break up text into more lines.

File: cves/CVE-2016-8740.yml [hunk 4]
```diff
@@ -98,8 +110,12 @@ unit_tested:
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
   answer:
+    The last test was written 5 years prior to this error, and most other tests were written over 10 years
+    prior. this was not properly tested nor was it updated following the error,
   code:
+    false
   fix:
+    false
```
Comment (mev1482) [inline cves/CVE-2016-8740.yml:118]: Capitalize this. False should be same line not new line

File: cves/CVE-2016-8740.yml [hunk 5]
```diff
@@ -222,3 +254,7 @@ mistakes:
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
   answer:
+    One of the biggest mistake is the lack of integration/unit tests
+    that verify for strange inputs and weird situations. In addition, better coding
+    practices like quality assurance testing would go a long way in resolving this
+    issue.
```
Comment (mev1482) [inline cves/CVE-2016-8740.yml]: Beyond testing they might have implemented better design to check input size. I agree that unit testing and QA would have helped to find the problem after its injection into the system.

**Review by mev1482 — CHANGES_REQUESTED** [review]

File: cves/CVE-2016-8740.yml (hunk 2 above)
Comment (17mgeffert) [reply cves/CVE-2016-8740.yml]: good idea, ill be sure to include it

File: cves/CVE-2016-8740.yml (hunk 4 above)
Comment (17mgeffert) [reply cves/CVE-2016-8740.yml:118]: just for consistency, I like to keep all text on new line, in addition, as false it technically not a string, but really a boolean, i think lowercase false is fine.

File: cves/CVE-2016-8740.yml (hunk 5 above)
Comment (17mgeffert) [reply cves/CVE-2016-8740.yml]: good catch

**Pull request merged**